vllm/entrypoints/llm.py

```diff
@@ -189,6 +189,93 @@ def generate(
 
         return self._run_engine(use_tqdm)
 
+    def chat(
+        self,
+        messages: Union[str, List[str], List[Dict[str, str]],
+                        List[List[Dict[str, str]]]],
+        sampling_params: Optional[Union[SamplingParams,
+                                        List[SamplingParams]]] = None,
+        use_tqdm: bool = True,
+        lora_request: Optional[LoRARequest] = None,
+        global_system_prompt: Optional[str] = None,
+        chat_template: Optional[str] = None,
+    ) -> List[RequestOutput]:
+        """
+        Generates responses for chat messages.
+
+        Converts the messages to prompts using the tokenizer and calls 
+        the `generate` method to generate the responses.
+
+        Args:
+            messages: A list of messages to generate responses for. Each 
+                message is a list of dictionaries with 'role' and 'content' 
+                keys.
+            sampling_params: The sampling parameters for text generation. 
+                If None, we use the default sampling parameters. When it 
+                is a single value, it is applied to every prompt. When it 
+                is a list, the list must have the same length as the 
+                prompts and it is paired one by one with the prompt.
+            use_tqdm: Whether to use tqdm to display the progress bar.
+            lora_request: LoRA request to use for generation, if any.
+            global_system_prompt: The initial system prompt used for each 
+            single turn conversation.
+            chat_template: The template to use for structuring the chat.
+              If not provided, the model's default chat template will be used.
+
+        Returns:
+            A list of `RequestOutput` objects containing the generated 
+            responses in the same order as the input messages.
+        """
+
+        tokenizer = self.get_tokenizer()
+        default_system_prompt = "You are a helpful assistant."
+
+        if isinstance(messages, str):
+            # Single turn conversation
+            messages = [{
+                'role':
+                'system',
+                'content':
+                global_system_prompt or default_system_prompt
+            }, {
```

Comment:
some model (like Mixtral) doesn't support system prompt

Reply:
Could you provide some guidance on how to handle these models? Is there a method or property I can check to see if a model supports system prompts?